homeassistant/components/sunsynk/sensor.py

```diff
@@ -0,0 +1,166 @@
+        native_unit_of_measurement=UnitOfPower.WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        name="Battery Level",
+        key=BATTERY_SOC,
+        device_class=SensorDeviceClass.BATTERY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+    SensorEntityDescription(
+        name="Solar Power",
+        key=SOLAR_POWER,
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement=UnitOfPower.WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        name="Solar Energy Today",
+        key=SOLAR_ENERGY_TODAY,
+        icon="mdi:solar-power",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Solar Energy Total",
+        key=SOLAR_ENERGY_TOTAL,
+        icon="mdi:solar-power",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Grid Import Today",
+        key=GRID_ENERGY_IMPORT_TODAY,
+        icon="mdi:power-plug",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Grid Import Total",
+        key=GRID_ENERGY_IMPORT_TOTAL,
+        icon="mdi:power-plug",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Grid Export Today",
+        key=GRID_ENERGY_EXPORT_TODAY,
+        icon="mdi:power-plug",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        name="Grid Export Total",
+        key=GRID_ENERGY_EXPORT_TOTAL,
+        icon="mdi:power-plug",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a Sunsynk entry."""
+    inverter_sn = entry.data[DATA_INVERTER_SN]
+
+    coordinator: SunsynkCoordinator = hass.data[DOMAIN][entry.entry_id][
+        SUNSYNK_COORDINATOR
+    ]
+
+    sensors = [
+        SunsynkSensor(coordinator, description, inverter_sn)
+        for description in SENSOR_TYPES
+    ]
+
+    async_add_entities(sensors)
+
+
+class SunsynkSensor(CoordinatorEntity[SunsynkCoordinator], SensorEntity):
```

Comment:
New integrations have to use `has_entity_name = True`, for more info checkout https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations

Reply:
Thanks, done